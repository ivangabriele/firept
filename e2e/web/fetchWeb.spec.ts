import got from 'got'
// import nock from 'nock';
// import app from '../../src/index'

describe('E2E tests', () => {
  // Mock Brave Search API response
  // beforeAll(() => {
  //   nock('https://api.search.brave.com').get('/res/v1/web/search').query(true).reply(200, {
  //     // Your mock response here
  //   })
  // })

  // it('should fetch search results from /web/search', async () => {
  //   const query = 'test query';
  //   const res = await request(app.callback()).get(`/web/search?query=${query}`);

  //   expect(res.status).toBe(200);
  //   expect(res.body).toHaveProperty('searchResults');
  // });

  it('should fetch and convert web page from /web/fetch', async () => {
    const url = 'https://example.org'
    const res = await got.get(`http://localhost:3333/web/fetch?url=${url}`)

    expect(res.ok).toBe(true)
    expect(res.body).toMatchSnapshot()
  })
})
