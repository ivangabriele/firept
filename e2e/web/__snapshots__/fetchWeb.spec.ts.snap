// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E tests should fetch and convert web page from /web/fetch 1`] = `"{"sourceAsMarkdown":"# Example Domain\\n\\nThis domain is for use in illustrative examples in documents. You may use this domain in literature without prior coordination or asking for permission.\\n\\n[More information...](https://www.iana.org/domains/example)"}"`;
